strongbox-storage/strongbox-storage-api/src/main/java/org/carlspring/strongbox/artifact/archive/Bzip2ArchiveListingFunction.java

```diff
@@ -0,0 +1,44 @@
+package org.carlspring.strongbox.artifact.archive;
+
+import org.carlspring.strongbox.providers.io.RepositoryPath;
+
+import java.io.BufferedInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Set;
+
+import org.apache.commons.compress.archivers.ArchiveInputStream;
+import org.apache.commons.compress.archivers.tar.TarArchiveInputStream;
+import org.apache.commons.compress.compressors.bzip2.BZip2CompressorInputStream;
+
+/**
+ * @author Przemyslaw Fusik
+ */
+public enum Bzip2ArchiveListingFunction
+        implements ArchiveListingFunction
+{
+
+    INSTANCE;
+
+    @Override
+    public Set<String> listFilenames(final RepositoryPath path)
+            throws IOException
+    {
+        try (InputStream is = Files.newInputStream(path);
+             BufferedInputStream bis = new BufferedInputStream(is);
+             BZip2CompressorInputStream bzIs = new BZip2CompressorInputStream(bis);
+             ArchiveInputStream tarIs = new TarArchiveInputStream(bzIs))
+        {
+            return getEntriesNames(tarIs);
+        }
+    }
+
+    @Override
+    public boolean supports(final RepositoryPath path)
+    {
+        final Path fileName = path.getFileName();
+        return fileName != null && fileName.toString().endsWith(".tar.bz2");
```

Comment:
Cosmetic: Not that it makes a difference, but the extension should not start with a dot. Would you mind changing this?

Reply:
sure

Reply:
Please be aware that there are also other places in the code like this. See for example `org.carlspring.strongbox.providers.layout.NugetLayoutProvider#isArtifactMetadata` or `org.carlspring.strongbox.providers.layout.Maven2LayoutProvider#isArtifactMetadata` or `org.carlspring.strongbox.cron.services.impl.CronJobSchedulerServiceImpl#getGroovyScriptsName`, ...